compiler/optimizer/LoopReducer.cpp

```diff
@@ -1088,23 +1088,24 @@ TR_LoopReducer::generateArraycopy(TR_InductionVariable * indVar, TR::Block * loo
    if (arraycopyLoop.getStoreAddress()->getIncrement() < 0)
       //&& (storeAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol() == loadAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol()))
       {
-      if (comp()->cg()->getSupportsPrimitiveArrayCopy())
-         arraycopy->setBackwardArrayCopy(true); /* bit available only to primitive arraycopy */
+      arraycopy->setBackwardArrayCopy(true);
       }
    else
       {
       arraycopy->setForwardArrayCopy(true);
       }
 
-   if (!comp()->cg()->getSupportsPrimitiveArrayCopy()) /* bits only available for non-primitive arraycopy */
+   switch (arraycopyLoop.getCopySize())
       {
-      switch(arraycopyLoop.getCopySize())
-         {
-         case 2: arraycopy->setHalfWordElementArrayCopy(true); break;
-         case 4: case 8: arraycopy->setWordElementArrayCopy(true); break;
-         }
-      }
+      case 2:
+         arraycopy->setHalfWordElementArrayCopy(true);
+         break;
 
+      case 4:
+      case 8:
+         arraycopy->setWordElementArrayCopy(true);
+         break;
+      }
```

Comment:
These also seem to be safe to set regardless of the value of `getSupportsPrimitiveArrayCopy`. They will still be set on the platforms that query them, and they'll be ignored on platforms where they weren't previously set.

The availability of the flag bits cannot depend on whether the particular `arraycopy` node is for copying primitives, since nothing here queries the node about that.

The flag bits are also available on all platforms. One of the bits collides with `dontTransformArrayCopyCall`, but we already better not transform a call node into `arraycopy` in-place when that flag is set.